agent/hcp/deps.go

```diff
@@ -27,15 +27,17 @@ type Deps struct {
 func NewDeps(cfg config.CloudConfig, logger hclog.Logger, nodeID types.NodeID) (d Deps, err error) {
```

Comment:
it doesn't look like we throw errors here if `hcpclient.NewClient` or `scada.New` fail, we just log. I'll admit I'm not terribly familiar with how these dependencies get wired in so I'm just double checking: is that expected?

Reply:
yeah so this was previously there already,  but it's using the Go pattern where you can define the `err` in the return params.. Named return values. https://go.dev/tour/basics/7, so `(d Deps, err error)`

This will return an error if client fails or scada fails, but not for the sink.

I'll just change this change this in a follow-up commit, as I think it's confusing for readers too.

Reply:
I'm familiar with named return values but I think this might have a bug as written because we might fail to make an `hcpclient.Client` but successfully make a `scada` client and overwrite the `err` with nil so not return `err`

Reply:
Ahhh you're right, there's a return. I missed the return buried between the comments 👍